Change to First-Person and Player Can Aim Where To Shoot Bullets

jamilfaisal opened the pull request:
- Changed camera to first-person view
- Player can now aim turret with mouse, arrow keys, or controller left-stick

jamilfaisal committed:
- Change camera to first-person using Cinemachine
- Player can aim with mouse (need to implement using new input system)

jamilfaisal committed:
Improve camera and shooting by pivoting the camera with the turret. Move cinemachine outside of Player gameobject

jamilfaisal committed:
- Add new Unity Input system packages
- Allow player to control camera with mouse, controller, and arrow keys using new input system

jamilfaisal committed:
Merge branch 'master' of https://github.com/jamilfaisal/Apocalypse-Bros into Change-to-first-person-and-aim

# Conflicts:
#	Assets/Prefabs/Player1.prefab
#	Assets/Scenes/Map1.unity
#	Assets/Scripts/Player/PlayerShoot.cs
#	Assets/Settings/PlayerAttributes.asset

CooperativeDev commented on Assets/Scripts/Player/PlayerLook.cs:
```diff
@@ -0,0 +1,49 @@
+            _mouseSensitivity = sensitivity;
+        }
+    }
+}
```
bro, no new line? really dude? ever heard of convention? doesn't even save noticeable space (by end user) smh.

jamilfaisal replied:
sorry massa

CooperativeDev commented on Assets/Scripts/Player/PlayerLook.cs:
```diff
@@ -0,0 +1,49 @@
+        private void Start()
+        {
+            _playerInputActions = GetComponent<PlayerInput>().GetPlayerInputActions();
+            Debug.Log(_playerInputActions);
```
REMOVE DEBUG LOG.

jamilfaisal replied:
oh mb

CooperativeDev commented on Assets/Scripts/Player/PlayerLook.cs:
```diff
@@ -0,0 +1,49 @@
+        private void RotateVertically(float mouseY)
+        {
+            _verticalRotation -= mouseY;
+            _verticalRotation = Mathf.Clamp(_verticalRotation, -45f, 20f);
```
-45f, 20f should be constants

jamilfaisal replied:
done sir

CooperativeDev commented on Assets/Scripts/Player/PlayerLook.cs:
```diff
@@ -0,0 +1,49 @@
+            _verticalRotation -= mouseY;
+            _verticalRotation = Mathf.Clamp(_verticalRotation, -45f, 20f);
+            virtualCamera.transform.localRotation = Quaternion.Euler(_verticalRotation, 0f, 0f);
+            turret.transform.localRotation = Quaternion.Euler(_verticalRotation + 90, 0f, 0f);
```
why +90?

jamilfaisal replied:
The turret's rotation.x is at 90. I added a parent turret object to separate the logic from visuals. This +90 isn't needed anymore

CooperativeDev commented on Assets/Scripts/Player/PlayerShoot.cs:
```diff
@@ -43,7 +43,8 @@ private bool ShootTimer(double reloadRate)
         private void ShootBullet(int damage)
         {
-            var bullet = Instantiate(bulletPrefab, gunEnd.position, gunEnd.rotation);
+            var gunEnd = new Vector3(gun.position.x, gun.position.y, gun.position.z);
```
why make a new vector 3 gunEnd, why not keep it the same as before (var bullet = Instantiate(bulletPrefab, gun.position, gun.rotation);)

jamilfaisal replied:
removed

CooperativeDev reviewed (changes requested):
dont forget to add line at the bottom of your files in future scripts

jamilfaisal committed:
- Add newline at the end of PlayerLook.cs
- Remove Debug.Log
- Separate turret visual from logic to remove the +90 from the vertical rotation calculation. Adjust min and max vertical look clamp values
- Convert min and max vertical look clamp values to constants
- Fix redundant var gunEnd variable creation in PlayerShoot.cs

jamilfaisal committed:
Remove double mouseSensitivity modifier for vertical rotation of camera

CooperativeDev reviewed (approved):
good work bro, gj
